test/runtests.jl

```diff
@@ -219,11 +219,32 @@ const BT = BinaryTrees
     @test BT.value(BT.search(tree, (0, 0, 1))) == 1
     @test BT.value(BT.search(tree, (1, 0, 0))) == 3
 
+    # traversal algorithms
+    tree = AVLTree{Int,Float64}()
+    BT.insert!(tree, 0, 5)
+    BT.insert!(tree, 1, 6)
+    BT.insert!(tree, 2, 8)
+    BT.insert!(tree, 3, 10)
+    BT.insert!(tree, 4, 20)
+    BT.insert!(tree, 5, 30)
+    BT.insert!(tree, 6, 40)
+    @test BT.key(BT.minnode(tree)) == 0
+    @test BT.key(BT.maxnode(tree)) == 6
+    @test BT.abovebelow(tree, BT.AVLNode(0, 5))[2] == nothing
+    @test BT.key.(BT.abovebelow(tree, BT.AVLNode(2, 10))) == (3, 1)
+    @test BT.key.(BT.abovebelow(BT.root(tree), BT.AVLNode(2, 10))) == (3, 1)
+    @test BT.key.(BT.abovebelow(tree, BT.AVLNode(5, 30))) == (6, 4)
+    @test BT.abovebelow(tree, nothing) == (nothing, nothing)
+    @test BT.abovebelow(BT.root(tree), nothing) == (nothing, nothing)
+
     # type stability
     tree = AVLTree{Int,Int}()
     @inferred BT.insert!(tree, 2, 20)
     @inferred BT.insert!(tree, 1, 10)
     @inferred BT.insert!(tree, 3, 30)
+    @inferred BT.minnode(tree)
+    @inferred BT.maxnode(tree)
```

Comment:
```suggestion
    @inferred Nothing BT.minnode(tree)
    @inferred Nothing BT.maxnode(tree)
```
Add `Nothing` as an allowed type.
Check the `@inferred` docstring for more details:

![image](https://github.com/user-attachments/assets/18b25ec8-a53a-414f-88de-b0e8086a9230)